pkg/kv/kvserver/replica_init.go

```diff
@@ -37,18 +37,27 @@ const (
 	mergeQueueThrottleDuration = 5 * time.Second
 )
 
+// useDefaultConnectionClassForRaft is a cluster setting to control whether
+// the default rpc.ConnectionClass should be used for Raft traffic or not. In
+// versions >= 24.1 there is an EnvVar for this purpose.
+var useDefaultConnectionClassForRaft = settings.RegisterBoolSetting(
+	settings.SystemOnly,
+	"kv.raft.use_default_connection_class",
+	"if true, use the default rpc.ConnectionClass for raft traffic instead of the raft-specific one",
+	true, // default 23.2 behaviour
+)
+
 // defRaftConnClass is the default rpc.ConnectionClass used for non-system raft
-// traffic. Normally it is RaftClass, but can be flipped to DefaultClass if the
-// corresponding env variable is true.
+// traffic. Normally it is DefaultClass, but can be flipped to RaftClass if the
+// corresponding cluster setting is true.
```

Comment:
nit: the cluster setting should be false. May just say "can be flipped to RaftClass with the cluster setting above".